### DISCUSSION TITLE
Support JsonEncodedText propertyName for JsonElement.TryGetProperty

### DISCUSSION THREAD

**Discussion by campersau** [dotnet/runtime#53855] [Ideas] [answered]
Is there a specific reason why `System.Text.Json.JsonElement.TryGetProperty` does not accept a `JsonEncodedText` as a propertyname parameter directly? It would make it easier to use the existing `JsonEncodedText`s for the `Utf8JsonWriter` to be also used by `JsonElement.TryGetProperty`.

The current workaround is to pass in the `JsonEncodedText.EncodedUtf8Bytes` manually like this:
```c#
System.Text.Json.JsonEncodedText nameJson = System.Text.Json.JsonEncodedText.Encode("name");
System.Text.Json.JsonDocument doc = ...;

doc.RootElement.TryGetProperty(nameJson.EncodedUtf8Bytes, out System.Text.Json.JsonElement _);
```

Suggestion:

```diff
namespace System.Text.Json
{
    public readonly struct JsonElement
    {
        // existing methods
        public bool TryGetProperty(ReadOnlySpan<byte> utf8PropertyName, out JsonElement value);
        public bool TryGetProperty(ReadOnlySpan<char> propertyName, out JsonElement value);
        public bool TryGetProperty(string propertyName, out JsonElement value);

+       public bool TryGetProperty(JsonEncodedText propertyName, out JsonElement value);
    }
}
```

The same question also applies to `System.Text.JsonUtf8JsonReader.ValueTextEquals`.

**Comment by huoyaoyuan** [comment, marked as answer]
This is an api suggestion and is better to be an issue.

**Reply by campersau** [reply]
Okay, done in https://github.com/dotnet/runtime/issues/53858
